ui/src/components/Tasks/TeamActivity/index.tsx

```diff
@@ -118,30 +133,57 @@ const ActivitiesRow: React.FC<ActivitiesRowProps> = (p) => {
 
   return (
     <>
-      <div className={c({ skeleton: p.isFetching })}>
-        <div className={c('container')}>
-          <div className={c('user-container')}>
-            <div className={c('user-logo')}>
-              <FallbackIcon logo={p.logo} name={p.title} />
-            </div>
-            <span className={c('user-info')}>
-              <span>{p.title}</span>
-              <Tippy content={dayjs(p.subtitle).format('DD, MMM YYYY hh:mm a')}>
-                <span className={c('task-time')}>{date}</span>
-              </Tippy>
-            </span>
+      <div className={c('container')}>
+        <div className={c('user-container')}>
+          <div className={c('user-logo')}>
+            <FallbackIcon logo={p.logo} name={p.title} />
           </div>
-
-          <div className={c('activites')}>{tasks}</div>
+          <span className={c('user-info')}>
+            <span>{p.title}</span>
+            <Tippy content={dayjs(p.subtitle).format('DD, MMM YYYY hh:mm a')}>
+              <span className={c('task-time')}>{date}</span>
+            </Tippy>
+          </span>
         </div>
+
+        <div className={c('activites')}>{tasks}</div>
       </div>
     </>
   );
 };
 
 const TeamActivity: React.FC = () => {
-  const activities = useSelector(activitySelector.selectAll);
-  const [isFetching] = useQuery(fetchTaskActivites, {});
+  const [pagination, setPagination] = useState({ limit: 20, offset: 0 });
+  const allActivities = useSelector(activitySelector.selectAll);
+
+  const activities = allActivities.slice(0, pagination.offset + pagination.limit);
+  const hasNext = allActivities.length <= pagination.offset + pagination.limit;
+
+  const [isFetching, getTaskActivities] = useQuery(
+    (args = { offset: pagination.offset, limit: pagination.limit + 1 }) => fetchTaskActivites(args),
+    {},
+  );
+
+  const onPaginationChange = () => {
+    const newOffset = pagination.offset + pagination.limit;
+    setPagination({ ...pagination, offset: newOffset });
+
+    if (!hasNext) {
+      getTaskActivities({ offset: newOffset, limit: pagination.limit + 1 });
+    }
+
+    return;
+  };
+
+  useEffect(() => {
+    const reFecth = setInterval(() => {
+      onPaginationChange();
+    }, 600000);
```

Comment:
This is a huge number. Leftover code perhaps?
Also, it's better to explicitly call out when a variable is holding reference to an interval; i.e naming it `refetchInterval` or `reFetchInterval` will be better imo.